lib/Elastica/Reindex.php

```diff
@@ -0,0 +1,117 @@
+<?php
+namespace Elastica;
+
+use Elastica\Query\AbstractQuery;
+
+class Reindex
+{
+    const VERSION_TYPE = 'version_type';
+    const VERSION_TYPE_INTERNAL = 'internal';
+    const VERSION_TYPE_EXTERNAL = 'external';
+    const OPERATION_TYPE = 'op_type';
+    const OPERATION_TYPE_CREATE = 'create';
+    const CONFLICTS = 'conflicts';
+    const CONFLICTS_PROCEED = 'proceed';
+    const TYPE = 'type';
+    const SIZE = 'size';
+    const QUERY = 'query';
+
+    public static function reindex(Index $oldIndex, Index $newIndex, array $options = [])
```

Comment:
I think that call `new` and `->run()` (or `->start()`) are ever near to constructor and there is no need to change options after have called `new Reindex(...)`. Maybe in a first time we can see how community uses reindex and if there will be need to change the use, we will change it consequently.